designs/dragonphy_top/constraints/gen_constraints.py

```diff
@@ -32,19 +33,83 @@
     -waveform {{0 {0.5*main_per*time_scale}}} \\
     [get_pins iacore/clk_adc]
 
-# TX clocks
-# TODO: add generated clocks in TX
+#############
+# TX clocks #
+#############
 
-create_clock -name clk_prbsgen \\
-    -period {main_per*time_scale} \\
-    -waveform {{0 {0.5*main_per*time_scale}}} \\
-    [get_pins itx/clk_prbsgen]
+# PI outputs
+
+create_clock -name clk_tx_pi_0 \\
+    -period {clk_4x_per} \\
+    -waveform {{0 {0.5*clk_4x_per}}} \\
+    [get_pins itx/iPI[0].iPI/clk_out_slice]
+
+create_clock -name clk_tx_pi_1 \\
+    -period {clk_4x_per} \\
+    -waveform {{{0.25*clk_4x_per} {0.75*clk_4x_per}}} \\
+    [get_pins itx/iPI[1].iPI/clk_out_slice]
+
+create_clock -name clk_tx_pi_2 \\
+    -period {clk_4x_per} \\
+    -waveform {{{0.5*clk_4x_per} {clk_4x_per}}} \\
+    [get_pins itx/iPI[2].iPI/clk_out_slice]
+
+create_clock -name clk_tx_pi_3 \\
+    -period {clk_4x_per} \\
+    -waveform {{{0.75*clk_4x_per} {1.25*clk_4x_per}}} \\
+    [get_pins itx/iPI[3].iPI/clk_out_slice]
 
-# clock uncertainty
+# clock dividers
```

Comment:
This is where the behavior of the internal clock dividers in the TX is declared.